chromium/external/README.md

```diff
@@ -0,0 +1,42 @@
+# External dependencies
+
+This directory contains files that are external dependencies for HTTPS Everywhere, with instructions on how to get or remake them.
+
+## CodeMirror 5.31.0
+
+```
+$ npm install uglify-js@3.1.9
```

Comment:
Since this is not a package dependency and is intended to be run as a command line app, let's add a `-g` flag for global install as the installation instructions specify: https://github.com/mishoo/UglifyJS2#install

Reply:
What's the drawback of defining all the dependencies in `package.json` in desired versions and define npm scripts that would execute proper commands?

Global packages should be avoided when possible. Someone may skip those steps as they've already performed them before and not notice they've since installed different versions since some other project required different ones. Using local packages gets rid of that problem.

Reply:
Good point, but the above command will cause `uglify-js` to be added to the `dependencies` hash, rather than the `devDependencies` hash.  Thus we should preference running this with the `-D` flag, and the below reproducibility instructions should be modified to be:
`../node_modules/uglify-js/bin/uglifyjs`

Reply:
Is there any reason to specify those packages directly instead of just having `package.json` checked into the repo and running `npm install`?

BTW, if you're referring to the binary by path, it's usually better to use `../node_modules/.bin/uglifyjs` as that depends less on the internal package structure. `node_modules/.bin` contains links to binaries exported by top-level packages.

Reply:
@mgol creating a dependency management system is on the table, but I don't want to block merging until we have one in place.  See https://github.com/EFForg/https-everywhere/pull/13560#issuecomment-344083403 and resulting discussion.

Reply:
These instructions are supposed to give someone who is knowledgeable about `npm` the key information needed to reproduce the files: package names, version numbers, options used. It's not supposed to completely walk a beginner through the process of making these files. The user can decide for themselves whether to install the minifier executables globally or not and how to get the files into whatever directories they want.